soc/arm/microchip_mec/mec172x/reg/mec172x_regs.h

```diff
@@ -0,0 +1,1119 @@
+/*
+ * Copyright (c) 2009-2018 Arm Limited. All rights reserved.
+ * Copyright (c) 2021 Microchip Technology Inc. and its subsidiaries.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MEC172X_REGS_H
+#define MEC172X_REGS_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * MEC172x includes the ARM single precicision FPU and the ARM MPU with
+ * eight regions.
+ */
+#define __MPU_PRESENT		1
+#define __FPU_PRESENT		1
+
+struct btmr_regs {
```

Comment:
missing comment?

Reply:
Hi @sjg20 I moved these to soc.h and added comments explaining why they must be set.  The other EC vendor you are using with Zephyr does the same.  Zephyr has a cmsis.h header that includes the proper core header from hal_microchip. Zephyr's cmsis.h does not set all the special __XXXX defines ARM CMSIS requires.